File: FE/FPT-EduTrack/src/enum/examStatus.ts
/**
 * Kỳ thi đang diễn ra, học sinh/sinh viên đang làm bài
 * Kỳ thi đã hoàn tất, không còn hoạt động thi cử nào
 * Bài thi đang được chấm điểm bởi ban tổ chức
 * Kết quả kỳ thi đã được công bố cho học sinh/sinh viên
 * Học sinh/sinh viên đã yêu cầu xem xét lại điểm số, và ban tổ chức đang xử lý các yêu cầu này
 * Kỳ thi đã bị hoãn hoặc được lên lịch lại do một số lý do (thời tiết, vấn đề kỹ thuật, v.v.)
 * Kỳ thi đã bị hủy hoàn toàn và sẽ không được tổ chức
 */
export type ExamStatus =
  | "InProgress"
  | "Completed"
  | "Grading"
  | "ResultsPublished"
  | "UnderReview"
  | "Postponed"
  | "Cancelled";

export const ExamStatus = {
  InProgress: "InProgress" as ExamStatus,
  Completed: "Completed" as ExamStatus,
  Grading: "Grading" as ExamStatus,
  ResultsPublished: "ResultsPublished" as ExamStatus,
  UnderReview: "UnderReview" as ExamStatus,
  Postponed: "Postponed" as ExamStatus,
  Cancelled: "Cancelled" as ExamStatus,
};

File: BE/FPT_EduTrack/FPT_EduTrack.BusinessLayer/DTOs/Request/ExamStatus.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FPT_EduTrack.BusinessLayer.DTOs.Request
{
    public enum ExamStatus
    {
        /// <summary>
        /// Exam is currently ongoing, students are taking the exam
        /// </summary>
        InProgress,

        /// <summary>
        /// Exam has been completed, no more examination activities
        /// </summary>
        Completed,

        /// <summary>
        /// Exam papers are being graded by the organizing committee
        /// </summary>
        Grading,

        /// <summary>
        /// Exam results have been published to students
        /// </summary>
        ResultsPublished,

        /// <summary>
        /// Students have requested score review, and the organizing committee is processing appeals
        /// </summary>
        UnderReview,

        /// <summary>
        /// Exam has been postponed or rescheduled due to some reason (weather, technical issues, etc.)
        /// </summary>
        Postponed,

        /// <summary>
        /// Exam has been completely cancelled and will not be held
        /// </summary>
        Cancelled
    }
}
